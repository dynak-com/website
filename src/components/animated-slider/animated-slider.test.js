import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import AnimatedSlider from './animated-slider.component';

describe('AnimatedSlider component', () => {
    const component = shallow(<AnimatedSlider />);
    const wrapper = mount(<AnimatedSlider />);
    const animatedSlider = component.find('div');

    it('renders correctly according to Snapshot', () => {
        const tree = renderer.create(<AnimatedSlider />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render 2 divs', () => {
        expect(component.find('div')).toHaveLength(2);
    });

    it('AnimatedSlider has 1 nested child', () => {
        expect(wrapper.children().length).toBe(1);
    });

    it('animatedSlider div has 1 child', () => {
        const imageContainer = animatedSlider.find('div');
        expect(imageContainer.children().length).toBe(1);
    });
});
