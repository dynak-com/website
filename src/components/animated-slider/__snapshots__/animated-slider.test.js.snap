// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedSlider component AnimatedSlider has 1 nested child 1`] = `ShallowWrapper {}`;

exports[`AnimatedSlider component animatedSlider div has 1 child 1`] = `ShallowWrapper {}`;

exports[`AnimatedSlider component renders correctly according to Snapshot 1`] = `
<div
  className="animatedSlider"
>
  <div
    className="imageContainer"
  />
</div>
`;
